src/Symfony/Component/PropertyInfo/Extractors/PhpDocExtractor.php

```diff
@@ -0,0 +1,392 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyInfo\Extractors;
+
+use phpDocumentor\Reflection\ClassReflector;
+use phpDocumentor\Reflection\DocBlock;
+use phpDocumentor\Reflection\FileReflector;
+use Symfony\Component\PropertyInfo\PropertyDescriptionInfoInterface;
+use Symfony\Component\PropertyInfo\PropertyTypeInfoInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * Extracts data using a PHPDoc parser.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class PhpDocExtractor implements PropertyDescriptionInfoInterface, PropertyTypeInfoInterface
+{
+    const PROPERTY = 0;
+    const ACCESSOR = 1;
+    const MUTATOR = 2;
+
+    /**
+     * @var FileReflector[]
+     */
+    private static $fileReflectors = array();
+    /**
+     * @var DocBlock[]
+     */
+    private static $docBlocks = array();
```

Comment:
I'm not a fan of using static properties to store them: we are leaking a global state here, not releasing memory.